### PR TITLE
refactor: Centralizing sigmoid implementation (FX Converter Refactor [2/N]) <Target: converter_reorg_proto>

### PR DESCRIPTION
**Pull request by narendasan**
# Description

Centralizes the sigmoid converters into one implementation

Fixes # (issue)

## Type of change

Please delete options that are not relevant and/or add your own.

- Refactor

# Checklist:

- [x] My code follows the style guidelines of this project (You can use the linters)
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas and hacks
- [x] I have made corresponding changes to the documentation
- [x] I have added tests to verify my fix or my feature
- [x] New and existing unit tests pass locally with my changes
- [x] I have added the relevant labels to my PR in so that relevant reviewers are notified

### CODE REVIEW

File: py/torch_tensorrt/fx/test/converters/aten_op/test_sigmoid_aten.py [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+                return nn.functional.sigmoid(x)
+
+        input_specs = [
+            InputTensorSpec(
+                shape=(-1, -1, -1, -1),
+                dtype=torch.float32,
+                shape_ranges=[((1, 1, 1, 5), (1, 2, 3, 5), (3, 3, 3, 5))],
+            ),
+        ]
+
```
Comment (wushirong) [inline py/torch_tensorrt/fx/test/converters/aten_op/test_sigmoid_aten.py:46]: can you add a unit test for float16? thanks

**Comment by narendasan** [comment]
It is not removed randomly. All checks have been centralized in the core function instead of making the responsibility that of each converter. See the first PR

From: wushirong ***@***.***>
Date: Thursday, May 4, 2023 at 3:16 PM
To: pytorch/TensorRT ***@***.***>
Cc: Naren Dasan ***@***.***>, Author ***@***.***>
Subject: Re: [pytorch/TensorRT] refactor: Centralizing sigmoid implementation (FX Converter Refactor [2/N]) (PR #1868)

@wushirong commented on this pull request.

________________________________

In py/torch_tensorrt/fx/converters/acc_ops_converters.py<https://github.com/pytorch/TensorRT/pull/1868#discussion_r1185550801>:

> -    if not isinstance(input_val, TRTTensor):

-        raise RuntimeError(

-            f"Sigmoid received input {input_val} that is not part "

-            "of the TensorRT region!"

-        )

Please do not remove the condition check randomly.

________________________________

In py/torch_tensorrt/fx/test/converters/aten_op/test_sigmoid_aten.py<https://github.com/pytorch/TensorRT/pull/1868#discussion_r1185556960>:

> +            TestModule(), input_specs, expected_ops={torch.ops.aten.sigmoid.default}

+        )

+

+    def test_sigmoid_with_dynamic_shape_four_dimensions(self):

+        class TestModule(nn.Module):

+            def forward(self, x):

+                return nn.functional.sigmoid(x)

+

+        input_specs = [

+            InputTensorSpec(

+                shape=(-1, -1, -1, -1),

+                dtype=torch.float32,

+                shape_ranges=[((1, 1, 1, 5), (1, 2, 3, 5), (3, 3, 3, 5))],

+            ),

+        ]

+

can you add a unit test for float16? thanks

File: py/torch_tensorrt/fx/converters/acc_ops_converters.py [hunk 2]
```diff
@@ -3192,21 +3192,13 @@ def acc_ops_sigmoid(
     kwargs: Dict[str, Argument],
     name: str,
 ) -> Union[TRTTensor, Sequence[TRTTensor]]:
-    input_val = kwargs["input"]
 
-    if not isinstance(input_val, TRTTensor):
-        raise RuntimeError(
-            f"Sigmoid received input {input_val} that is not part "
-            "of the TensorRT region!"
-        )
```
[the remark this answers is not in the document]
Comment (narendasan) [reply py/torch_tensorrt/fx/converters/acc_ops_converters.py:3201]: It is not removed randomly. Tech checks have been centralized in the core function instead of making the responsibility that of each converter. See the first PR #1867

[the remark this answers is not in the document]
Comment (wushirong) [reply py/torch_tensorrt/fx/converters/acc_ops_converters.py:3201]: The movement change in #1867 lgtm, left comment on the other pr.

**Comment by narendasan** [comment]
@wushirong Can you review this PR. This is gating release of 1.4?

File: py/torch_tensorrt/fx/test/converters/aten_op/test_sigmoid_aten.py (hunk 1 above)
Comment (wushirong) [reply py/torch_tensorrt/fx/test/converters/aten_op/test_sigmoid_aten.py:46]: bump on this comment.

**Comment by wushirong** [comment]
can you check the comment on the adding fb16 test cases? once we get this, I think the diff is good to go.
